endpoints/v2/blob.py

```diff
@@ -62,6 +63,12 @@ def check_blob_exists(namespace_name, repo_name, digest):
     # Find the blob.
     blob = registry_model.get_cached_repo_blob(model_cache, namespace_name, repo_name, digest)
     if blob is None:
+        try:
+            proxy = setup_proxy(namespace_name, repo_name)
+            resp = proxy.blob_exists(digest)
+            return Response(**resp)
+        except ProxyNotSupported as e:
+            logger.debug(f"Skipping pull through proxy cache: {e}")
```

Comment:
We talked about this last week - using a decorator as you suggest assumes we _want_ all the proxy code in the same place, and I honestly can't say we do (or do not) at this point. I will not refactor code just to refactor it again using a different approach in the future. 
Can focus on the _what_ rather than on the _how_, at least for this first iteration?
Most of the view code in this PR will change in the near future, it's not worth spending either your or my time trying to find the correct abstraction at this point.